frontend/src/components/SignUp/SignUpForm.jsx

```diff
@@ -0,0 +1,159 @@
+import React from "react";
+
+import TextField from '@material-ui/core/TextField';
+import Button from '@material-ui/core/Button';
+import Grid from '@material-ui/core/Grid';
+import Card from '@material-ui/core/Card';
+import CardContent from '@material-ui/core/CardContent';
+import Typography from '@material-ui/core/Typography';
+
+import { withStyles } from '@material-ui/core/styles';
+
+import 'typeface-roboto';
+
+const styles = theme => ({
+  form: {
+    marginTop: 50,
+  },
+  cardBody: {
+    padding: 20,
+  },
+  textField: {
+    marginLeft: theme.spacing.unit,
+    marginRight: theme.spacing.unit,
+    width: 300,
+  },
+  button: {
+    margin: theme.spacing.unit,
+  },
+});
+
+class SubmittedForm extends React.Component {
+  render() {
+    const {classes} = this.props;
+    return (
+      <div>
+        <Grid container justify='center' className={classes.form}>
+          <Typography variant="h4" align='center'>
+            {this.props.message}
+          </Typography>
+          <Grid container direction='column' alignContent='center'>
+            <Grid item>
+              <Button variant="outlined" href="/" className={classes.button}>
+                Go to homepage
+              </Button>
+            </Grid>
+          </Grid>
+        </Grid>
+      </div>
+    );
+  }
+}
+
+class SignUpForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: "",
+      email: "",
+      password: "",
+      serverResponse: false,
+      errors: false
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(e) {
+    this.setState({[e.target.name]: e.target.value});
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    const {name, email, password} = this.state;
+    const formData = {name, email, password};
```

Comment:
Client side validation is needed here. With clear messages what is wrong.

Reply:
Done. Review, please.